File: frontend/src/app/shared/model/ui/configuration.ts
import { ConfigurationId } from '../id-types';

export class Configuration {
  id: ConfigurationId;
  name: string;
  value: string | boolean;
  type: string;

  constructor(id: ConfigurationId, name: string, value: string | boolean, type: string) {
    this.id = id;
    this.name = name;
    this.value = value;
    this.type = type;
  }
}

File: backend/burning-okr/burning-okr-model/src/main/java/org/burningokr/model/configuration/Configuration.java
package org.burningokr.model.configuration;

import jakarta.persistence.*;
import lombok.Data;
import org.burningokr.model.activity.Trackable;

@Entity
@Data
public class Configuration implements Trackable<Long> {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO, generator = "hibernate_sequence_generator")
  @SequenceGenerator(name = "hibernate_sequence_generator", sequenceName = "hibernate_sequence", allocationSize = 1)
  private Long id;

  private String name;

  private String value;

  private String type;
}
